scripts/zones/Castle_Zvahl_Baileys/npcs/Switchstix.lua

```diff
@@ -9,291 +9,303 @@ require("scripts/globals/keyitems")
-    18297, {16834, 16836, 16841,  4,  1456,  16,  2,  2},
-    18298, { 1563,     0,     0,  2,  1453,  61,  3,  3},
-    18299, { 1578,  1589,     0,  0,  1451,   1,  4,  3},
-
-    18302, { 1458,  1117, 13208,  4,  1450,   5,  1,  1}, -- Apocalypse
-    18303, {16774, 16794, 16777,  4,  1453,  16,  2,  2},
-    18304, { 1562,     0,     0,  2,  1456,  62,  3,  3},
-    18305, { 1577,  1589,     0,  0,  1451,   1,  4,  3},
-
-    18308, { 1467,  1276,  1278,  4,  1456,   4,  1,  1}, -- Kikoku
-    18309, {16900, 16903, 16902,  4,  1450,  16,  2,  2},
-    18310, { 1564,     0,     0,  2,  1453,  61,  3,  3},
-    18311, { 1579,  1589,     0,  0,  1457,   1,  4,  3},
-
-    18314, { 1467,  1409,   657,  4,  1453,   3,  1,  1}, -- Amanomurakumo
-    18315, {16966, 16967, 16972,  4,  1450,  15,  2,  2},
-    18316, { 1565,     0,     0,  2,  1456,  60,  3,  3},
-    18317, { 1580,  1589,     0,  0,  1454,   1,  4,  3},
-
-    18320, { 1461,   746,   830,  4,  1453,   5,  1,  1}, -- Mjollnir
-    18321, {17044, 17080, 17043,  4,  1456,  16,  2,  2},
-    18322, { 1566,     0,     0,  2,  1450,  62,  3,  3},
-    18323, { 1581,  1589,     0,  0,  1454,   1,  4,  3},
-
-    18326, { 1462,  1271,  1415,  4,  1450,   5,  1,  1}, -- Claustrum
-    18327, {17088, 17090, 17092,  4,  1456,  16,  2,  2},
-    18328, { 1567,     0,     0,  2,  1453,  62,  3,  3},
-    18329, { 1582,  1589,     0,  0,  1451,   1,  4,  3},
-
-    18332, { 1468,   830,   654,  4,  1456,   5,  1,  1}, -- Annihilator
-    18333, {17248, 17251, 17259,  4,  1450,  15,  2,  2},
-    18334, { 1570,     0,     0,  2,  1453,  62,  3,  3},
-    18335, { 1585,  1589,     0,  0,  1457,   1,  4,  3},
-
-    18338, { 1458,  1463, 13232,  4,  1450,   3,  1,  1}, -- Gjallarhorn
-    18339, {17352, 17351, 17362,  4,  1456,  14,  2,  2},
-    18340, { 1569,     0,     0,  2,  1453,  60,  3,  3},
-    18341, { 1584,  1589,     0,  0,  1451,   1,  4,  3},
-
-    18344, {  883,  1462,   932,  4,  1453,   4,  1,  1}, -- Yoichinoyumi
-    18345, {17161, 17164, 18142,  4,  1453,  15,  2,  2},
-    18346, { 1568,     0,     0,  2,  1450,  61,  3,  3},
-    18347, { 1583,  1589,     0,  0,  1454,   1,  4,  3},
-
-    15066, {  875,   668,   720,  4,     0,   1,  1,  4}, -- Aegis
-    15067, {12301, 12295, 12387,  4,     0,   4,  2,  5},
-    15068, { 1821,     0,     0,  2,     0,  20,  3,  6},
-    15069, { 1822,  1589,     0,  0,  1454,   1,  4,  6}
+local requiredItems = 1
+local currencyType = 2
+local currencyAmount = 3
+local stageNumber = 4
+local csParam = 5
+
+local currency =
+{
+    BYNE_100 = 1456,
+    BYNE_10000 = 1457,
+    SILVER_100 = 1453,
+    SILVER_10000 = 1454,
+    SHELL_100 = 1450,
+    SHELL_10000 = 1451,
 }
 
-function hasRelic(entity, checktype)
-    -- Type 1 == Player is triggering NPC
-    -- Type 2 == Player is trading an item
+local relics =
+{
+    -- Spharai
+    [18260] = { { 1460, 1459, 665 }, currency.BYNE_100, 4, 1, 1 }, -- Relic Knuckles
```

Comment:
keys! the new table has keeeeeys!


Listen.. I love you. no homo.

Reply:
You should have seen me almost physically struggle with not creating _another_ new table to enumerate the relic weapon stage IDs into item names. 🙀